### DISCUSSION TITLE
Git Flow（ブランチの切り方について）

### DISCUSSION THREAD

**Discussion by satory074** [ShphieEnterprise/Sophie#12] [General]
今は`main`から`feature/#<refs>_<title>`でブランチを切っているかと思います。

以下、branch運用の改善案を提案しますのでご意見賜りたく。

1. `main`から`develop`を切り、`develoop`から`feature/#<refs>_<title>`を切る（`main`から直接ではなく）。
    > 機能開発後、`develop`にmergeし、動作検証後、`main`にmergeしてリリースする流れが良いのではないかと思ったため。

2. 障害時は`main`から`hotfix`を切って対応し、`main`にmergeする。

(`release`branchも一般的ですが、そこまでする必要はないのかなと思いました。)

### 参考

- https://dev.classmethod.jp/articles/introduce-git-flow/

----

## 提案するbranch運用
|||
|-|-|
|main|動作中のシステム|
|develop|`main`にmergeする前に、`feature`から`merge`したものを動作検証する用のbranch|
|feature|機能開発等用のbranch|
|hotfix|緊急対応用のbranch|

```mermaid
gitGraph
    commit tag: "0.1"
    branch hotfix order: 1
    branch develop order: 3
    commit

    branch feature_1 order: 4
    branch feature_2 order: 5
    commit

    checkout develop
    commit

    checkout feature_1
    commit

    checkout hotfix
    commit

    checkout main
    merge hotfix tag: "v0.2"

    checkout develop
    merge hotfix

    checkout feature_2
    commit

    checkout feature_1
    commit

    checkout develop
    merge feature_1

    checkout feature_2
    commit

    checkout main
    merge develop tag: "v1.0"
    checkout develop
    merge main

    checkout feature_2
    commit

    checkout develop
    merge feature_2
```

**Comment by Cer0un0** [comment]
良いと思います

**Comment by satory074** [comment]
ありがとうございます
